dataland-frontend/src/api-models/ProcurementCategory.ts

```diff
@@ -0,0 +1,5 @@
+export enum ProcurementCategory {
+  Products = "Products",
+  RawMaterials = "RawMaterials",
+  Services = "Services",
+}
```

Comment:
Why not using the autogenerated enum from `ProcurementCategory.kt` ?

Reply:
xD I would absolutely love to. But there is no way to have it generated. It is not part of the openApiDocs due to some swagger bug.

Reply:
Wow ok that absolutely sucks.

But how come that other enums are generated without problems? E.g. `IdentifierType`, which is a backend enum model, and also available in the frontend.
Do you have clue?

Reply:
Ahh got it 👍 .     OpenApi converts  `Map<SomeEnumType, SomeKpis>` to `{ [key: string]: SomeKpis; };`  .
And therefore `ProcurementCategoryType.kt` is never actually used in the generated stuff, right?

OK will resolve this.

Reply:
Does this also mean, that we will never be able to use an object type as key to a map in our backend models, because openApi will do the conversion to a `string` that I just mentioned?

Reply:
I guess so. I think object valued keys are not possible here. How would that look in a JSON?

Reply:
Some research told me that it would look like this:
e.g. you have a `Map<Car, Owner>`  which maps cars to the respective owners of the car:

`
{
  "carOwners": {
    {
      "brand": "Toyota",
      "model": "Camry"
    }: {
      "name": "John Doe",
      "age": 30
    },
    {
      "brand": "Ford",
      "model": "Mustang"
    }: {
      "name": "Jane Smith",
      "age": 35
    }
  }
}`

Reply:
But nevermind, I will close this conversation since openApi just does not generate the enum class we need and it is external software.